iroh-net/src/netcheck/reportgen.rs

```diff
@@ -1053,3 +1042,283 @@ async fn measure_https_latency(_node: &DerpNode) -> Result<(Duration, IpAddr)> {
     // // Maybe the server should return the tcpinfo_rtt?
     // return result.ServerProcessing, ip, nil
 }
+
+/// Updates a netcheck [`Report`] with a new [`ProbeReport`].
+fn update_report(report: &mut Report, probe_report: ProbeReport) {
+    let derp_node = probe_report.probe.node();
+    if let Some(latency) = probe_report.latency {
+        report
+            .derp_latency
+            .update_derp(derp_node.url.clone(), latency);
+
+        if matches!(
+            probe_report.probe.proto(),
+            ProbeProto::StunIpv4 | ProbeProto::StunIpv6
+        ) {
+            report.udp = true;
+
+            match probe_report.addr {
+                Some(ipp @ SocketAddr::V4(_)) => {
+                    report.ipv4 = true;
+                    report
+                        .derp_v4_latency
+                        .update_derp(derp_node.url.clone(), latency);
+                    if report.global_v4.is_none() {
+                        report.global_v4 = Some(ipp);
+                    } else if report.global_v4 != Some(ipp) {
+                        report.mapping_varies_by_dest_ip = Some(true);
+                    } else if report.mapping_varies_by_dest_ip.is_none() {
+                        report.mapping_varies_by_dest_ip = Some(false);
+                    }
+                }
+                Some(ipp @ SocketAddr::V6(_)) => {
+                    report.ipv6 = true;
+                    report
+                        .derp_v6_latency
+                        .update_derp(derp_node.url.clone(), latency);
+                    report.global_v6 = Some(ipp);
+                    // TODO: Should we track mapping_varies_by_dest_ip for IPv6 too?  Would
```

Comment:
would be interesting to track, and print a big log message if it happens, I am really curious if this happens in practice

Reply:
14ada614

Yeah, why not.  This was an old todo, possibly even copied from tailscale.  I assumed it meant a lot of changes, but it really isn't so added it.